client/src/pages/AccountSettingsPage.jsx

```diff
@@ -0,0 +1,320 @@
+  Card,
+  Grid,
+  Stack,
+  TextField,
+  Typography,
+} from '@mui/material';
+import React, { useEffect, useState } from 'react';
+
+import BackgroundShapes from '../components/BackgroundShapes';
+import CloseAccountButton from '../components/CloseAccountButton';
+import ErrorMessage from '../components/ErrorMessage';
+import SignOutButton from '../components/SignOutButton';
+import {
+  getUserAccountInfo,
+  updateUserEmail,
+  updateUserName,
+  updateUserPassword,
+} from '../utils/api';
+import { isEmailValid, isPasswordValid } from '../utils/helper';
+import BasePage from './BasePage';
+
+const AccountSettingsPage = () => {
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [email, setEmail] = useState('');
+  const [confirmEmail, setConfirmEmail] = useState('');
+  const [errorMessage, setErrorMessage] = useState(null);
+
+  const [editableFields, setEditableFields] = useState({
+    name: false,
+    password: false,
+    email: false,
+  });
+
+  const rowFields = [
+    {
+      editableFieldKey: 'name',
+      previewModeFields: {
+        label: 'Name',
+        value: firstName + ' ' + lastName,
+        buttonText: 'Change name',
+      },
+      editModeFields: {
+        fields: [
+          {
+            label: 'First name',
+            value: firstName,
+            isHidden: false,
+            onChange: (event) => setFirstName(event.target.value),
+          },
+          {
+            label: 'Last name',
+            value: lastName,
+            isHidden: false,
+            onChange: (event) => setLastName(event.target.value),
+          },
+        ],
+        buttonText: 'Save changes',
+        getErrorMessage: async () => {
+          try {
+            await updateUserName(firstName, lastName);
+          } catch {
+            return 'There was a problem updating your name. Please try again later';
+          }
+          return null;
+        },
+      },
+    },
+    {
+      editableFieldKey: 'password',
+      previewModeFields: {
+        label: 'Password',
+        value: '*'.repeat(password ? password.length : 0),
+        buttonText: 'Change password',
+      },
+      editModeFields: {
+        fields: [
+          {
+            label: 'Password',
+            value: password,
+            isHidden: true,
+            onChange: (event) => {
+              setPassword(event.target.value);
+            },
+          },
+          {
+            label: 'Confirm password',
+            value: confirmPassword,
+            isHidden: true,
+            onChange: (event) => setConfirmPassword(event.target.value),
+          },
+        ],
+        buttonText: 'Save password',
+        getErrorMessage: async () => {
+          if (confirmPassword !== password) {
+            return 'Your password and confirmation password must match.';
```

Comment:
To make the function syntactically nicer, i think it would be better to throw the error messages instead of return them. Then you could get rid of the return null. (follow up in lower comment in the onClick)